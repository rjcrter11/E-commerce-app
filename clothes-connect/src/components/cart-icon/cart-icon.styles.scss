.cart-icon {
     
   
    width: 45px;
    height: 50px; 
    position: relative;
    display: flex; 
    align-items: center;
    justify-content: center;
    cursor: pointer;
    margin-bottom: 2px; 

    .shopping-icon {
        width: 26px; 
        height: 26px; 
    }

    .item-count {
        position: absolute;
        font-size: 10px;
        font-weight: bold; 
        bottom: 14px; 
    }

}