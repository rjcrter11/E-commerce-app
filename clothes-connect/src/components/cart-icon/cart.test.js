import React from 'react';
import { shallow } from 'enzyme';
import { CartIcon } from './cart-icon.component';


describe('CartIcon Component', () => {
    let wrapper;
    let mockToggleCartHidden;

    beforeEach(() => {
        mockToggleCartHidden = jest.fn();

        const mockProps = {
            itemCount: 0,
            mockToggleCartHidden: mockToggleCartHidden
        };

        wrapper = shallow(<CartIcon {...mockProps} />)
    });

    it('should render CartIcon Component', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('should call toggleCartHidden when icon is clicked', () => {
        wrapper.find('[data-testid="cart-icon"]').simulate('click');
        expect(mockToggleCartHidden).toHaveBeenCalledTimes(0);
    })
    it('should render the itemCount as the text', () => {
        const itemCount = parseInt(wrapper.find('[data-testid="item-count"]').text());
        expect(itemCount).toBe(0)
    })
})
