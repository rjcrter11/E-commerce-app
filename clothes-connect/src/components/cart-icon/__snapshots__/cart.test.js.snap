// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon Component should render CartIcon Component 1`] = `ShallowWrapper {}`;
